File: messages/src/main/java/com/thundashop/zauiactivity/dto/OctoProductAvailabilityRequestDto.java
package com.thundashop.zauiactivity.dto;

import java.util.List;

import lombok.Data;

@Data
public class OctoProductAvailabilityRequestDto {
    private Integer productId;
    private String optionId;
    private String localDateStart;
    private String localDateEnd;
    private String currency;
    private List<AvailabilityUnit> units = null;
}
class AvailabilityUnit {
    String id;
    Integer quantity;
}

File: com.getshop.client/events/zauiactivity/dto/OctoProductAvailabilityRequestDto.php
<?php
class zauiactivity_dto_OctoProductAvailabilityRequestDto {
	/** @var String */
	public $productId;

	/** @var String */
	public $optionId;

	/** @var String */
	public $localDateStart;

	/** @var String */
	public $localDateEnd;

	/** @var String */
	public $currency;

	/** @var class com_thundashop_zauiactivity_dto_AvailabilityUnit[] */
	public $units;

}
?>